lib/galaxy/schema/invocation.py

```diff
@@ -218,3 +253,272 @@ class InvocationMessageResponseModel(BaseModel):
+        title="Markdown",
+        description="Raw galaxy-flavored markdown contents of the report.",
+    )
+    model_class: schema.INVOCATION_REPORT_MODEL_CLASS = ModelClassField(schema.INVOCATION_REPORT_MODEL_CLASS)
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Workflow ID",
+        description="The workflow this invocation has been triggered for.",
+    )
+    username: str = Field(
+        ...,
+        title="Username",
+        description="The name of the user who owns this report.",
+    )
+    title: str = Field(
+        ...,
+        title="Title",
+        description="The name of the report.",
+    )
+    generate_time: Optional[str] = schema.GenerateTimeField
+    generate_version: Optional[str] = schema.GenerateVersionField
+
+    class Config:
+        # Galaxy Report/Page response can contain many extra_rendering_data
+        # Allow any other extra fields
+        extra = Extra.allow
+
+
+class InvocationUpdatePayload(Model):
+    action: bool = InvocationStepActionField
+
+
+class InvocationIOBase(Model):
+    id: EncodedDatabaseIdField = Field(
+        default=Required, title="ID", description="The encoded ID of the dataset/dataset collection."
+    )
+    workflow_step_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Workflow step ID",
+        description="The encoded ID of the workflow step associated with the dataset/dataset collection.",
+    )
+
+
+class InvocationInput(InvocationIOBase):
+    label: Optional[str] = Field(
+        default=Required,
+        title="Label",
+        description="Label of the workflow step associated with the input dataset/dataset collection.",
+    )
+    src: Union[Literal[DataItemSourceType.hda], Literal[DataItemSourceType.hdca]] = Field(
+        default=Required, title="Source", description="Source type of the input dataset/dataset collection."
+    )
+
+
+class InvocationInputParameter(Model):
+    # TODO - Change the type of parameter_value, when all valid types are known
+    parameter_value: Any = Field(default=Required, title="Parameter value", description="Value of the input parameter.")
+    label: str = Field(
+        default=Required, title="Label", description="Label of the workflow step associated with the input parameter."
+    )
+    workflow_step_id: EncodedDatabaseIdField = Field(
+        default=Required,
+        title="Workflow step ID",
+        description="The encoded ID of the workflow step associated with the input parameter.",
+    )
+
+
+class InvocationOutput(InvocationIOBase):
+    src: Literal[DataItemSourceType.hda] = Field(
+        default=Required, title="Source", description="Source model of the output dataset."
+    )
+
+
+class InvocationOutputCollection(InvocationIOBase):
+    src: Literal[DataItemSourceType.hdca] = Field(
+        default=Required, title="Source", description="Source model of the output dataset collection."
+    )
+
+
+class WorkflowInvocationResponse(BaseModel):
```

Comment:
```suggestion
class WorkflowInvocationResponse(Model):
```

In general, you want to make your models inherit from `schema.Model` and not directly from Pydantic `BaseModel`, the reason is, that our custom `Model` has some special features, for example, if you use `ModelClassField` it will only work for our custom base model, hence the OpenAPI linting error you are getting.

https://github.com/galaxyproject/galaxy/blob/f10614ee48532a10f3aa353ab40b409647dfc763/lib/galaxy/schema/schema.py#L280